tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -211,12 +213,23 @@ function configure_containerd() {
 
 function configure_kata() {
 	if [ "${CONFIGURE_CC:-}" == "yes" ]; then
+		cc_config="$kata_defaults_dir/configuration-cc.toml"
 		sed -E \
-			-e 's#^image = .+#initrd = "/opt/kata/share/kata-containers/kata-containers-initrd.img"#' \
-			-e 's#^(kernel_params = .+)"#\1 agent.config_file=/etc/kata-config.toml"#' \
-			-e 's#.*service_offload = .+#service_offload = true#' \
-			"/opt/kata/share/defaults/kata-containers/configuration-qemu.toml" > \
-			"/opt/kata/share/defaults/kata-containers/configuration-cc.toml"
+			-e 's|^(kernel_params = .+)"|\1 agent.config_file=/etc/kata-containers/agent.toml"|' \
+			-e 's|^[# ]*(service_offload = ).+|\1true|' \
+			"$kata_defaults_dir/configuration-qemu.toml" > \
+			"$cc_config"
+		if [ "$(uname -m)" == "s390x" ]; then
+			sed -Ei \
+				-e "s|^(kernel = ).+|\\1\"$kata_share_dir/kata-containers-secure.img\"|" \
+				-e '/^image = .+/d' \
+				-e 's|^[# ]*(confidential_guest = ).+|\1true|' \
```

Comment:
Thanks @Jakob-Naucke. My only question from an x86 perspective is if this confidential_guest flag is s390x specific, or if it could apply for the entire CONFIGURE_CC case?

Reply:
@Alex-Carter01 it isn't, but your question got me thinking. When enabled, the use of the platform-specific confidential computing technology is attempted. I hadn't enabled this in general, because on x86, if you're not capable of TDX/SEV, there will be errors, and this script is also supposed to work for a generic try-out setup without a real TEE. Always enabling it on s390x is a slight oversimplification (although TBF no one really has _just_ s390x available); ideally, we'd run a Secure Execution check here. However, we're already checking this [in the runtime](https://github.com/kata-containers/kata-containers/blob/d370604fa5cdc78ecc438ee3c65e2e805c54c160/src/runtime/virtcontainers/hypervisor_s390x.go) _plus_ there's https://github.com/confidential-containers/operator/issues/24 which also seems to want to deal with this.

@bpradipt DYT it is unnecessary to check capabilities in operator, kata-deploy, _and_ runtime?

Reply:
I see, thank you for the clarification. It makes sense to me then why it is right to leave it here, and then have an equivalent for x86 behind a Secure Execution check if that ends up being necessary.

Reply:
@Jakob-Naucke checking of capabilities in operator is required as we'll be supporting confidential computing technologies and it's important to have the visibility at the operator layer for life-cycle management of the runtimes. A simple use-case being installing the kata-deploy-cc artefacts for SEV only on capable systems in the cluster.
It's also good to have the capabilities check in the runtime since there will be use-cases where the runtime is executed on single system (without requiring k8s+operator infra).
As for kata-deploy, I'm not sure if having specific CC capability checks will help. For CC case, I would prefer keeping kata-deploy as pure payload, and having the capability check in the operator and the runtime.
For example kata-deploy image for CC on s390x having all required configs for SE  (quay.io/confidential-containers/runtime-payload-s390x). Likewise for SEV, TDX etc.

Reply:
@Jakob-Naucke following is the high level workflow of the operator:
1. Deploy operator
2. Create CC config. This will detect the eligibility of the nodes in the cluster w.r.to specific TEE. For example if it's an s390x cluster, then it will detect which nodes in the cluster are eligible to run confidential containers using SE. Likewise for SEV and TDX _(note that this functionality is not yet implemented)_
3. Once the operator knows the eligible nodes, it'll perform the required installation tasks on the specific nodes using the kata-deploy payload. Current implementation sets the CONFIGURE_CC env variable and execute the installer script.

Having the CC settings under a single conditional (`CONFIGURE_CC`) is better.  You can as well add CC capability check in kata-deploy in case CONFIGURE_CC is unset, but honestly I'm not sure if it's really necessary to add additional CC checks to kata-deploy. 

Whatever other combinations are needed to determine a true value for `CONFIGURE_CC` can be handled by the operator.

Reply:
Thanks @bpradipt. To summarise:
- _For the time being_, on x86_64, `kata-deploy.sh` will install the non-TEE "CC" setup. (In the longer term, we might want to drop that overall maybe? We should avoid anyone running real workloads with the non-TEE demo thinking they had enabled TEE.)
- On s390x, `kata-deploy.sh` will already assume it was invoked on a capable machine.

Reply:
@Jakob-Naucke yes. Once we have SEV and TDX pieces we'll have a real TEE "CC" setup and can rename the non-TEE "CC" setup accordingly (test/demo etc..)

Reply:
Rather than the arch test, you could use the following (which needs an update for `s390x` I think?):

```bash
$ kata-runtime env --json | jq '.Host.AvailableGuestProtections'
```

Reply:
Ah -- I could not reply further down because you can only reply in the original thread, which is here.
My original comment (which I've deleted) was:

> > Rather than the arch test, you could use the following (which needs an update for s390x I think?):
> > ```
> > $ kata-runtime env --json | jq '.Host.AvailableGuestProtections'
> > ```
> 
> This does work with Secure Execution, the only trouble is that this wants `vmlinux.container` and `kata-containers.img`. I'll look into it.

and to give an update on that: it does work when setting using the CC config, but that introduces a self-reference because that config is only generated here. So I'd keep it.